massa-execution-worker/src/execution.rs

```diff
@@ -1053,6 +1076,12 @@ impl ExecutionState {
                 .expect("Missing block in storage.")
                 .clone();
 
+            block_info = Some(BlockInfo {
+                block_id: *block_id,
+                current_version: stored_block.content.header.content.current_version,
+                announced_version: stored_block.content.header.content.announced_version,
+            });
+
```

Comment:
have we tested that if someone announces a block with a current_version and/or a announced_version that our node does not know, we handle that well and cannot get attacked ?
* refuse blocks with an incorrect current_version
* ignore/warn on unknown announced_version
* make sure weird combinations (eg. someone announces an old version but sets current_version to the right one) are all handled correctly

Reply:
* Refuse blocks with an incorrect current_version:

This is done here:

https://github.com/massalabs/massa/blob/c0d97e2eec3645db4f221fa763566841d817a9b0/massa-protocol-worker/src/handlers/block_handler/retrieval.rs#L447-L468

For the 2 others points, here is a follow up:

* https://github.com/massalabs/massa/issues/4175